ssh.go

```diff
@@ -16,18 +16,21 @@ import (
 
 // Client is a wrapper over the SSH connection/sessions.
 type SSHClient struct {
-	conn         *ssh.Client
-	sess         *ssh.Session
-	user         string
-	host         string
-	remoteStdin  io.WriteCloser
-	remoteStdout io.Reader
-	remoteStderr io.Reader
-	connOpened   bool
-	sessOpened   bool
-	running      bool
-	env          string //export FOO="bar"; export BAR="baz";
-	color        string
+	conn               *ssh.Client
+	sess               *ssh.Session
+	user               string
+	host               string
+	sshKeys            []string // ssh key to use
+	remoteStdin        io.WriteCloser
+	remoteStdout       io.Reader
+	remoteStderr       io.Reader
+	connOpened         bool
+	sessOpened         bool
+	running            bool
+	env                string //export FOO="bar"; export BAR="baz";
+	color              string
+	initAuthMethodOnce sync.Once
```

Comment:
This might slow down the init phase of sup. Imagine running 100 remote SSH connections, you'd have to init Auth method 100 times. Do we really need that?

Imho, if the identity settings is per network, we don't need to do it per-host here, do we?